multiraft/multiraft.go

```diff
@@ -550,7 +571,43 @@ func (s *state) start() {
 
 				select {
 				case resp := <-s.writeTask.out:
-					s.handleWriteResponse(resp, writingGroups)
+					callbackCount := s.handleWriteResponse(resp, writingGroups)
+					// Wait till all callback returned and send buffered event if any.
```

Comment:
We should try not to have too much stuff in the `start()` method. Either move this into `handleWriteResponse` or make a new `waitForCallbacks` method.